crates/hotshot/src/lib.rs

```diff
@@ -602,8 +602,8 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
 
         if let Some(request_receiver) = da_network.spawn_request_receiver_task().await {
             add_response_task(&mut handle, request_receiver).await;
-            add_request_network_task(&mut handle).await;
         }
+        add_request_network_task(&mut handle).await;
```

Comment:
Ah it's so that the request for data call can fail and send something back to the caller for fetching proposal.  It simply returns an error if it's not implemented.  Otherwise we ended up waiting forever for a response.  This is a small hack to get it to work in the current impl that can probably be removed since we have the `async_timeout` but I still think it's best to fail faster.  I'll try to move it out so we don't end up getting random tries to request VID shares